repos/system_upgrade/common/actors/removeleftoverpackages/libraries/removeleftoverpackages.py

```diff
@@ -0,0 +1,58 @@
+from leapp.libraries.common.rhsm import skip_rhsm
+from leapp.libraries.common.rpms import get_installed_rpms
+from leapp.libraries.stdlib import api, CalledProcessError, run
+from leapp.models import LeftoverPackages, RemovedPackages, RPM
+
+
+def _get_leftover_packages():
+    """
+    Consume and return LefoverPackages, if there are none, return None
```

Comment:
Hmm do you really need `_get_leftover_packages` helper? Seems excessive as you are still exiting early if there are no packages, like in process
 
```
   leftover_packages = _get_leftover_packages()
    if leftover_packages is None:
        return
```
This can be simplified to 
```
leftover_packages = next(api.consume(LeftoverPackages), None)
if not leftover_packages:
    api.current_logger().info('No leftover packages, skipping...') 
    return
```

Reply:
Personally, I don't mind the more verbose style with the helper. It's a bit more self-documenting this way

Reply:
Im leaving as it is for now, if someone really wants it that way, can do it, with adapting tests to it